src/libraries/System.Private.CoreLib/src/System/Text/Decoder.cs

```diff
@@ -261,6 +261,13 @@ public virtual unsafe int GetChars(ReadOnlySpan<byte> bytes, Span<char> chars, b
 
             bytesUsed = byteCount;
 
+            if (byteCount == 0)
```

Comment:
Do we need to handle `flush: true` here? The encoder may have a pending state that needs to be flushed even when `byteCount` is zero.

Reply:
I was thinking that'd be covered by just reporting completed as false, but if that's not the semantic, sure, I can change it.

Reply:
The behavior with _byteCount = 0, flush = true_ should be that the decoder attempts to flush its internal state to the destination, and if the destination is not large enough to hold the entire result then the operation should fail with an exception.

This matches the philosophy of the Decoder / Encoder types: they always try to make some forward progress. If they cannot make any forward progress at all (due to the destination buffer being too small), they fail.

Reply:
I removed this commit and replaced it with one at the TranscodingStream level.  We can separately decide if/how to fix the issue in Encoder/Decoder.Convert.